### PR TITLE
feat(components, app): update Chip component for unification

### PR DESCRIPTION
**Pull request by koji**
# Overview
update Chip component for unification.
https://github.com/Opentrons/opentrons/assets/474225/0def2528-15b6-4425-971c-82303fcba5fe

a few tests are commented out because of jsdom's limitations (media-query). I will keep investigating this issue.

close AUTH-214

# Test Plan
- make teardown-js && make setup-js
- make -C components dev

# Changelog
- move Chip from app to components
- update Chip component, test, and storybook
- update all Chip import statements

# Review requests

# Risk assessment

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunRunTimeParameters.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { formatRunTimeParameterValue } from '@opentrons/shared-data'
 import {
   ALIGN_CENTER,
   BORDERS,
+  // Chip,
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunRunTimeParameters.tsx]: is there a reason why this is commented out?

File: components/src/atoms/Chip/index.tsx [hunk 2]
```diff
@@ -103,26 +100,54 @@ export function Chip(props: ChipProps): JSX.Element {
...
+  const TEXT_STYLE = css`
+    ${chipSize === 'medium' ? WEB_MEDIUM_TEXT_STYLE : WEB_SMALL_TEXT_STYLE}
+    @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+      ${chipSize === 'medium'
+        ? TYPOGRAPHY.bodyTextSemiBold
+        : TYPOGRAPHY.smallBodyTextSemiBold}
+    }
+  `
+
+  console.log(window.innerHeight, innerWidth)
```
Comment (jerader) [inline components/src/atoms/Chip/index.tsx]:
```suggestion
```

File: components/src/atoms/Chip/__tests__/Chip.test.tsx [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+}
+
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: vi.fn().mockImplementation(query => ({
+    matches: false,
+    media: '(height: 600px) and (width: 1024px)',
+    onchange: null,
+    addListener: vi.fn(), // Deprecated
+    removeListener: vi.fn(), // Deprecated
```
Comment (jerader) [inline components/src/atoms/Chip/__tests__/Chip.test.tsx]: do you need to add these if they are deprecated?

File: components/src/atoms/Chip/index.tsx [hunk 4]
```diff
@@ -103,26 +100,54 @@ export function Chip(props: ChipProps): JSX.Element {
...
+    @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+      padding: ${SPACING.spacing4}
+        ${background === false ? 0 : SPACING.spacing8};
+      grid-gap: ${SPACING.spacing4};
+    }
+  `
+
+  const ICON_STYLE = css`
+    width: ${chipSize === 'medium' ? '1rem' : '0.75rem'};
+    /* height: ${chipSize === 'medium' ? '1rem' : '0.75rem'}; */
```
Comment (jerader) [inline components/src/atoms/Chip/index.tsx]: why is height commented out?

**Review by jerader — COMMENTED** [review]
nice! i left a few comments

**Review by jerader — APPROVED** [review]
lgtm!

**Comment by koji** [comment]
@ncdiehl11, could you take a look, when you are available?

**Review by ncdiehl11 — APPROVED** [review]
This looks good. I can now use the unified `Chip` from `components` in my RTP PR.

**Pull request merged**